### PR TITLE
`namer` codelab use Dart 3 switch pattern

### PR DESCRIPTION
**Pull request by miquelbeltran**
Requires updating codelab document.

Only steps 07 and 08 affected.

## Pre-launch Checklist

- [ ] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [ ] I signed the [CLA].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by miquelbeltran** [comment]
@domesticmouse I think I'll need access to this codelab document to update the code snippets.

**Comment by miquelbeltran** [comment]
Thanks Filip! Yes, I agree that for this codelab is going to be too advanced to use switch patterns, so we can skip this one. I'll let Bret decide :)

On upgrading the codelab_rebuild.yaml, also true, I haven't done that on the other codelabs, but I'll have to start doing it as well.

**Comment by domesticmouse** [comment]
> Hi Miguel, I think you'll have to change the https://github.com/flutter/codelabs/blob/main/namer/codelab_rebuild.yaml file instead of the generated files. It's kind of a pain to do changes this way, tbh, but it makes sure future changes never really break the flow of codelabs.

Apologies for the pain, the upside is that I can update the codelabs for each stable release relatively painlessly.

**Comment by domesticmouse** [comment]
> On upgrading the codelab_rebuild.yaml, also true, I haven't done that on the other codelabs, but I'll have to start doing it as well.

I'm happy to deal with updating `codelab_rebuild.yaml`s. I'm sure Andrew would be happy to see people other than me dealing with this pain, but the upside is purely in the quarterly update cycle.

If we move to monthly stable releases, this will become doubly (triply?) important for my sanity. =)

**Comment by domesticmouse** [comment]
> Before you do that, though, I'd weigh whether we need to teach this new language construct to beginner Flutter developers. I definitely like your code better than the original, and it's what I would write working on a project, but in a beginner codelab, we have to be very cognizant of information overload.

I don't have strong feelings either way. I can see the upside for this change, but it'll need explanation added to a reasonably heavy getting started codelab. I think I'm leaning towards holding off making this change. 

Paging @RedBrogdon and @johnpryan for their input.

File: namer/step_07_d_use_selectedindex/lib/main.dart [hunk 1]
```diff
@@ -55,17 +55,11 @@ class _MyHomePageState extends State<MyHomePage> {
 
   @override
   Widget build(BuildContext context) {
-    Widget page;
-    switch (selectedIndex) {
-      case 0:
-        page = GeneratorPage();
-        break;
```
Comment (domesticmouse) [inline namer/step_07_d_use_selectedindex/lib/main.dart:62]: If we don't make this change, we should definitely lose the breaks.

Comment (filiph) [reply namer/step_07_d_use_selectedindex/lib/main.dart:62]: Agreed! The breaks should go.

**Comment by filiph** [comment]
FWIW, I would prefer that, yes.

**Comment by domesticmouse** [comment]
> One thing I was reminded yesterday, is that this codelab is also recorded on video, and so it would require to edit it if we do this type of change. I'm happy to close this one and submit one new PR without the `break` statements if you prefer that.

Sounds good to me, let's do that =)

**Comment by miquelbeltran** [comment]
All good! I will submit the PR shortly.

**Pull request closed without merging**